Maple2.Server.Game/Manager/BeautyManager.cs

```diff
@@ -0,0 +1,120 @@
+﻿using System;
+using System.Linq;
+using Maple2.Database.Storage;
+using Maple2.Model.Enum;
+using Maple2.Model.Error;
+using Maple2.Model.Game;
+using Maple2.Model.Metadata;
+using Maple2.Server.Game.Manager.Items;
+using Maple2.Server.Game.Packets;
+using Maple2.Server.Game.Session;
+using Serilog;
+
+namespace Maple2.Server.Game.Manager;
+
+public sealed class BeautyManager : IDisposable {
+    private readonly GameSession session;
+    private readonly ItemCollection items;
+    private short expand;
+    private Item? previousHair;
+    public BeautyManager(GameSession session) {
+        this.session = session;
+        items = new ItemCollection(Constant.BaseStorageCount);
+        using GameStorage.Request db = session.GameStorage.Context();
+        expand = db.GetHairStorageAmount(session.CharacterId);
+        foreach (Item item in db.GetSavedHairs(session.CharacterId).Where(item => items.Add(item).Count == 0)) {
+            Log.Error("Failed to add saved hair:{Uid}", item.Uid);
+        }
+    }
+
+    public void Dispose() {
+        using GameStorage.Request db = session.GameStorage.Context();
+        lock (session.Item) {
+            db.SaveItems(session.CharacterId, items.ToArray());
+        }
+    }
+
+    public void Load() {
+        lock (session.Item) {
+            session.Send(BeautyPacket.StartList());
+            session.Send(BeautyPacket.SaveSlots(expand));
+            session.Send(BeautyPacket.ListCount(items.Count));
+            if (items.Count > 0) {
+                session.Send(BeautyPacket.ListHair(items.OrderBy(hair => hair.CreationTime).ToList()));;
+            }
+        }
+    }
+
+    public void Add(long uid) {
```

Comment:
`BeautyManager.Add` is not clear. I think it should either be `BeautyManager.AddHair` or `BeautyHairManager.Add`